File: Week13/encoder/src/main/java/com/example/encoder/EncoderEndpoint.java
package com.example.encoder;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/encoder")
@RequiredArgsConstructor
public class EncoderEndpoint {

  private final Encoder encoder;

  @PostMapping
  String encode(@RequestBody String text) {
    return encoder.encode(text);
  }
}

File: Week9/morseencoderfromweek4/encoder/src/main/java/com/example/encoder/EncoderEndpoint.kt
import com.example.encoder.Encoder
import lombok.RequiredArgsConstructor
import org.springframework.web.bind.annotation .PostMapping
import org.springframework.web.bind.annotation .RequestBody
import org.springframework.web.bind.annotation .RequestMapping
import org.springframework.web.bind.annotation .RestController

@RestController
@RequestMapping("/encode")
@RequiredArgsConstructor
class EncoderEndpoint {

    private val encoder: Encoder? = null

    @PostMapping
    internal fun encode(@RequestBody message: String): String {
        return encoder!!.encode(message)
    }

}